src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final int FIGURE_COUNT = 5;
+    private static final int MIN_LENGTH = 2;
+    private static final int MAX_LENGTH = 10;
+    private static final int DEAFULT_FIGURE_RADIUS = 10;
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int ran = RANDOM.nextInt(FIGURE_COUNT);
+        int a = RANDOM.nextInt(MAX_LENGTH - MIN_LENGTH + 1) + MIN_LENGTH;
+        int b = RANDOM.nextInt(MAX_LENGTH - MIN_LENGTH + 1) + MIN_LENGTH;
+        int c = RANDOM.nextInt(MAX_LENGTH - MIN_LENGTH + 1) + MIN_LENGTH;
+        return switch (ran) {
+            case (0) -> new Square(a, colorSupplier.getRandomColor());
+            case (1) -> new Rectangle(a, b, colorSupplier.getRandomColor());
+            case (2) -> new RightTriangle(a, b, colorSupplier.getRandomColor());
+            case (3) -> new Circle(a, Color.WHITE);

```

Comment:
The color for the Circle in the random figure generation should be obtained from the ColorSupplier instead of being hardcoded to Color.WHITE.